flink-python/pyflink/datastream/connectors.py

```diff
@@ -0,0 +1,464 @@
+        records received by a sink subtask will end up in the same Kafka partition).
+
+        :param topic: ID of the Kafka topic.
+        :param serialization_schema: User defined key-less serialization schema.
+        :param producer_config: Properties with the producer configuration.
+        """
+        gateway = get_gateway()
+        j_properties = gateway.jvm.java.util.Properties()
+        for key, value in producer_config.items():
+            j_properties.setProperty(key, value)
+
+        JFlinkKafkaProducer010 = gateway.jvm \
+            .org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer010
+        j_flink_kafka_producer = JFlinkKafkaProducer010(
+            topic, serialization_schema._j_serialization_schema, j_properties)
+        super(FlinkKafkaProducer010, self).__init__(j_flink_kafka_producer=j_flink_kafka_producer)
+
+
+class Semantic(object):
+    """
+    Semantics that can be chosen.
+    :data: `EXACTLY_ONCE`:
+    The Flink producer will write all messages in a Kafka transaction that will be committed to
+    the Kafka on a checkpoint. In this mode FlinkKafkaProducer011 sets up a pool of
+    FlinkKafkaProducer. Between each checkpoint there is created new Kafka transaction, which is
+    being committed on FlinkKafkaProducer011.notifyCheckpointComplete(long). If checkpoint
+    complete notifications are running late, FlinkKafkaProducer011 can run out of
+    FlinkKafkaProducers in the pool. In that case any subsequent FlinkKafkaProducer011.snapshot-
+    State() requests will fail and the FlinkKafkaProducer011 will keep using the
+    FlinkKafkaProducer from previous checkpoint. To decrease chances of failing checkpoints
+    there are four options:
+        1. decrease number of max concurrent checkpoints
+        2. make checkpoints mre reliable (so that they complete faster)
+        3. increase delay between checkpoints
+        4. increase size of FlinkKafkaProducers pool
+
+    :data: `AT_LEAST_ONCE`:
+    The Flink producer will wait for all outstanding messages in the Kafka buffers to be
+    acknowledged by the Kafka producer on a checkpoint.
+
+    :data: `NONE`:
+    Means that nothing will be guaranteed. Messages can be lost and/or duplicated in case of
+    failure.
+    """
+
+    EXACTLY_ONCE = 0,
+    AT_LEAST_ONCE = 1,
+    NONE = 2
+
+    @staticmethod
+    def _to_j_semantic(semantic, j_semantic):
+        if semantic == Semantic.EXACTLY_ONCE:
+            return j_semantic.EXACTLY_ONCE
+        elif semantic == Semantic.AT_LEAST_ONCE:
+            return j_semantic.AT_LEAST_ONCE
+        elif semantic == Semantic.NONE:
+            return j_semantic.NONE
+        else:
+            raise TypeError("Unsupported semantic: %s, supported semantics are: "
+                            "Semantic.EXACTLY_ONCE, Semantic.AT_LEAST_ONCE, Semantic.NONE"
+                            % semantic)
+
+
+class FlinkKafkaProducer011(FlinkKafkaProducerBase):
```

Comment:
add `ignore_failures_after_transaction_timeout` in `FlinkKafkaProducer011 ` and `FlinkKafkaProducer `